src/rules/order.js

```diff
@@ -502,27 +502,34 @@ function makeNewlinesBetweenReport(context, imported, newlinesBetweenImports) {
 
     return linesBetweenImports.filter((line) => !line.trim().length).length;
   };
+  const getIsStartOfNaturalGroup = (currentImport, previousImport) => {
+    return currentImport.rank - 1 >= previousImport.rank;
+  };
   let previousImport = imported[0];
 
   imported.slice(1).forEach(function (currentImport) {
     const emptyLinesBetween = getNumberOfEmptyLinesBetween(currentImport, previousImport);
+    const isStartOfNaturalGroup = getIsStartOfNaturalGroup(currentImport, previousImport);
```

Comment:
i'm a bit confused by the "natural" terminology. what's natural about this, or unnatural about the alternative?

Reply:
I used "natural" because to me, it wasn't intuitive that setting a `position` in a `pathGroup` would appear to automatically create a new group, and add the newline. I assumed that only each `pathGroup` entry would potentially create a new group, and that by setting a `.position`, I would only be relatively positioning the matched imports _within_ that particular group. I felt that `pathGroupsNaturalNewlines: true` changed the behavior so that it felt more natural and expected.

Maybe the `natural` thing is too unclear - possibly the option can be changed to `patternPositionWithinGroup` and the function to something like `getShouldDivideGroup()`?

Reply:
hm - so you're saying that by default, if you *don't* set `position`, a pathGroup doesn't add an additional newline, but if you *do* set it, it does, unless this option is enabled?

Reply:
Not quite - I'm saying that by default, if you _don't_ set `position`, a `pathGroup` _does_ add an additional newline. But, if you _do_ set it, then _two_ newlines are added, unless this option is enabled

Reply:
two newlines, like next to each other? if so, that's a bug.

maybe it's just hard to tell in the diff (－‸ლ) could you maybe add two identical test cases, one with this option and one without, so i can see the difference? (or if that's already there, point me to it with my apologies)

Reply:
For sure, I've added a test case with identical imports for when this new option is `false`. Here are the permalinks

With [`pathGroupNaturalNewlines: false`](https://github.com/hyperupcall/eslint-plugin-import/blob/053e5716a90639a3cf9f46c528e832d4f82c4004/tests/src/rules/order.js#L928). This is the default case. Here, since `position` is set in `pathGroups[0]`, it creates a second newline after the `import A` line. Likewise, with `pathGroups[1]`, it creates a second newline after the `import C` line.

With [`pathGroupNaturalNewlines: true`](https://github.com/hyperupcall/eslint-plugin-import/blob/053e5716a90639a3cf9f46c528e832d4f82c4004/tests/src/rules/order.js#L958), for the same imports, no second newlines are created. This is where the "natural" comes in - both `'a'` and `'b'` are external imports, just like `c` - and intuitively, they are grouped together, and the `position` property positions them _within_ the `'external'` group, as expected

If this is a bug fix, rather than a feature request, maybe I should remove the option all together and just make it the regular behavior?